docs/core/extensions/httpclient-factory-keyed-di.md

```diff
@@ -0,0 +1,302 @@
+
+You don't have to call <xref:Microsoft.Extensions.DependencyInjection.HttpClientBuilderExtensions.AddAsKeyed%2A> for every single client&mdash;you can easily opt in "globally" (for any client name) via <xref:Microsoft.Extensions.DependencyInjection.HttpClientFactoryServiceCollectionExtensions.ConfigureHttpClientDefaults%2A>. From Keyed Services perspective, it results in the <xref:Microsoft.Extensions.DependencyInjection.KeyedService.AnyKey?displayProperty=nameWithType> registration.
+
+```csharp
+services.ConfigureHttpClientDefaults(b => b.AddAsKeyed());
+
+services.AddHttpClient("first", /* ... */);
+services.AddHttpClient("second", /* ... */);
+services.AddHttpClient("third", /* ... */);
+
+public class MyController(
+    [FromKeyedServices("first")] HttpClient first,
+    [FromKeyedServices("second")] HttpClient second,
+    [FromKeyedServices("third")] HttpClient third)
+//{ ...
+```
+
+### Beware Of "Unknown" clients
+
+> [!NOTE]
+> `KeyedService.AnyKey` registrations define a mapping from _any_ key value to some service instance. However, as a result, the Container validation doesn't apply, and an _erroneous_ key value _silently_ leads to a _wrong instance_ being injected.
+
+> [!IMPORTANT]
+> For Keyed `HttpClient`s, a mistake in the client name can result in erroneously injecting an "unknown" client&mdash;meaning, a client whose name was never registered.
+
+The same is true for the plain Named clients: `IHttpClientFactory` doesn't require the client name to be explicitly registered (aligning with the way the [Options pattern](options.md) works). The factory gives you an unconfigured&mdash;or, more precisely, default-configured&mdash;`HttpClient` for any unknown name.
+
+> [!NOTE]
+> Therefore, it's important to keep in mind: the "Keyed by default" approach covers not only all _registered_ `HttpClient`s, but all the clients that `IHttpClientFactory` is _able to create_.
+
+```csharp
+services.ConfigureHttpClientDefaults(b => b.AddAsKeyed());
+services.AddHttpClient("known", /* ... */);
+
+provider.GetRequiredKeyedService<HttpClient>("known");   // OK
+provider.GetRequiredKeyedService<HttpClient>("unknown"); // OK (unconfigured instance)
+```
+
+### "Opt-in" strategy considerations
+
+Even though the "global" opt-in is a one-liner, it's unfortunate that the feature still requires it, instead of just working "out of the box." For full context and reasoning on that decision, see [dotnet/runtime#89755](https://github.com/dotnet/runtime/issues/89755) and [dotnet/runtime#104943](https://github.com/dotnet/runtime/pull/104943). In short, the main blocker for "on by default" is the `ServiceLifetime` "controversy": for the current (`9.0.0`) state of the DI and `IHttpClientFactory` implementations, there's no single `ServiceLifetime` that would be reasonably safe for all `HttpClient`s in all possible situations. There's an intention, however, to address the caveats in the upcoming releases, and switch the strategy from "opt-in" to "opt-out".
```

Comment:
- Quotes for global are unnecessary. It's not so-called global, it is global.
- Remove unfortunate.